src/Symfony/Component/Process/ExecutableFinder.php

```diff
@@ -56,13 +56,20 @@ public function find($name, $default = null, array $extraDirs = array())
             $searchPath = explode(PATH_SEPARATOR, ini_get('open_basedir'));
             $dirs = array();
             foreach ($searchPath as $path) {
-                if (is_dir($path)) {
-                    $dirs[] = $path;
-                } else {
-                    if (basename($path) == $name && is_executable($path)) {
-                        return $path;
+                try {
+                    if (is_dir($path)) {
+                        $dirs[] = $path;
+                    } else {
+                        if (basename($path) == $name && is_executable($path)) {
+                            return $path;
+                        }
                     }
                 }
+                catch (\ErrorException $e) {
```

Comment:
is_dir triggers warnings, not exceptions. A custom error handler must be registered for an exception to be thrown.
The implementation with an `@` is required (same as @stof comments)